src/viam/app/data_client.py

```diff
@@ -453,6 +454,7 @@ async def binary_data_capture_upload(
             tags (Optional[List[str]]): Optional list of tags to allow for tag-based data filtering when retrieving data.
             data_request_times (Optional[Tuple[datetime.datetime, datetime.datetime]]): Optional tuple containing `datetime`s objects
                 denoting the times this data was requested[0] by the robot and received[1] from the appropriate sensor.
+            file_extension (str): The file extension of binary data including the period, e.g. .jpg, .png, .pcd
```

Comment:
Could you include that this routes the file to its corresponding mime type in the backend?

Reply:
@agreenb how's this for text:

> The backend will route the binary to its corresponding mime type based on this extension.

Reply:
Perfect, thanks!